cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -593,11 +599,35 @@ void sc_mount_nvidia_driver(const char *rootfs_dir)
 		die("cannot change ownership of " SC_LIB);
 	}
 	(void)sc_set_effective_identity(old);
+
+	/* We include the globs for the glvnd libraries for old snaps
+	 * based on core, Ubuntu 16.04 did not include glvnd itself.
+	 *
+	 * While there is no guarantee that the host system's glvnd
+	 * libGL will be compatible (as it is built with the host
+	 * system's glibc), the Mesa libGL included with the snap will
+	 * definitely not be compatible (as it expects to find the Mesa
+	 * implementation of the GLX extension)..
+	 */
+	const char **globs = nvidia_globs;
+	size_t globs_len = nvidia_globs_len;
+	const char **full_globs SC_CLEANUP(sc_cleanup_shallow_strv) = NULL;
+	if (sc_streq(base_snap_name, "core")) {
```

Comment:
Since there are no unit tests for this part of s-c code, I think that the spread test should be extended to cover an old core based snapd an one that's using core18 or core20. Probably best to just drop the use of test-snapd-policy-app-consumer from the test and add 2 snaps for it.